Make text-orientation:sideways-right an alias for sideways in Gecko

mbrubeck opened the pull request:
Fixes #15214. r? @upsuper

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/15268)
<!-- Reviewable:end -->

highfive commented:
Heads up! This PR modifies the following files:
 * @bholley: components/style/properties/properties.mako.rs, components/style/properties/gecko.mako.rs, components/style/properties/helpers.mako.rs, components/style/properties/longhand/inherited_box.mako.rs, components/style/properties/data.py
 * @emilio: components/style/properties/properties.mako.rs, components/style/properties/gecko.mako.rs, components/style/properties/helpers.mako.rs, components/style/properties/longhand/inherited_box.mako.rs, components/style/properties/data.py

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify style code, but no tests are modified. Please consider adding a test!

jdm commented on components/style/properties/data.py:
```diff
@@ -48,6 +49,7 @@ def __init__(self, name, values, gecko_constant_prefix=None,
         self.gecko_enum_prefix = gecko_enum_prefix
         self.extra_gecko_values = (extra_gecko_values or "").split()
         self.extra_servo_values = (extra_servo_values or "").split()
+        self.gecko_alias_values = {} if gecko_alias_values is None else gecko_alias_values
```
This is equivalent to `gecko_alias_values or {}`.

mbrubeck replied:
Fixed.

upsuper commented on components/style/properties/properties.mako.rs:
```diff
@@ -1676,6 +1675,7 @@ pub fn get_writing_mode(inheritedbox_style: &style_structs::InheritedBox) -> Wri
         computed_values::text_orientation::T::mixed => {},
         computed_values::text_orientation::T::upright => {},
     % endif
+        computed_values::text_orientation::T::sideways_right => {},
```
If `sideways-right` is an alias of `sideways`, why do they have different effect here?

Gecko drops `sideways-right` value (parses it as `sideways`), which I guess Servo can do as well. Is there any reason to keep Servo's `sideways-right` separate from `sideways`?

mbrubeck committed:
Convert text-orientation to a Gecko-only property

Remove incomplete and buggy support for text-orientation in Servo.
Make the property values align with Gecko and the latest draft of CSS
Writing Modes Level 3.

mbrubeck commented:
Servo layout does not actually support `text-orientation` at all, so I updated this patch to make it a Gecko-only property for now.

upsuper reviewed (approved):
LGTM

Wafflespeanut commented:
The unit tests seem to fail.

mbrubeck commented:
@bors-servo r=upsuper

* Pushed a fix for the unit tests.

bors-servo commented:
:pushpin: Commit b3820c1 has been approved by `upsuper`

<!-- @bors-servo r=upsuper b3820c1296a4a9703de81acb62fcf1d1cfbd8a62 -->

bors-servo commented:
:hourglass: Testing commit b3820c1 with merge 9ace55b...

bors-servo commented:
:broken_heart: Test failed - [windows-gnu-dev](http://build.servo.org/builders/windows-gnu-dev/builds/559)

mbrubeck commented:
@bors-servo retry

bors-servo commented:
:zap: Previous build results for [android](http://build.servo.org/builders/android/builds/5101), [arm32](http://build.servo.org/builders/arm32/builds/5281), [arm64](http://build.servo.org/builders/arm64/builds/5321), [linux-dev](http://build.servo.org/builders/linux-dev/builds/5425), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/2106), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/2118), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/5400), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/5439), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/2096), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/2150) are reusable. Rebuilding only [windows-gnu-dev](http://build.servo.org/builders/windows-gnu-dev/builds/559), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/564)...

bors-servo commented:
:broken_heart: Test failed - [windows-gnu-dev](http://build.servo.org/builders/windows-gnu-dev/builds/568)

mbrubeck commented:
@bors-servo retry

bors-servo commented:
:hourglass: Testing commit b3820c1 with merge ec12b76...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/5115), [arm32](http://build.servo.org/builders/arm32/builds/5298), [arm64](http://build.servo.org/builders/arm64/builds/5338), [linux-dev](http://build.servo.org/builders/linux-dev/builds/5442), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/2123), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/2137), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/5418), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/5456), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/2113), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/2168), [windows-gnu-dev](http://build.servo.org/builders/windows-gnu-dev/builds/578), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/584)
